### PR TITLE
Add button that dismisses all notifications

### LINKED ISSUE #4983: Dismiss All button in persistent notification sidebar
## The request

There are scenarios where it's possible to get a LOT of persistent notifications if you're not careful with an automation. For example, you add a new RSS feed to the feedreader integration and it pulls in every feed item and notifies you. Or even the "Motion Detected" notification example on the demo site.

Dismissing dozens of these notifications one by one is extremely tedious. So how about a "Dismiss All" button either at the top or bottom, that you could press to quickly get rid of them all? Other platforms like iOS and Android have this feature and it comes in handy for managing notifications. Could make this only show up when there's >2 notifications or something.

dismiss

### PR DESCRIPTION
**Pull request by mattmattmatt**
## Proposed change

Resolves [this feature request](https://github.com/home-assistant/frontend/issues/4983) that proposes a Dismiss All button in the notification list.

The new functionality adds a button at the bottom of the notification list that is only visible when there are 2 or more notifications. When pressed, a loop calls the `persistent_notification.dismiss` service on each notification, after which the notification drawer closes.

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: closes #4983
- This PR is related to issue:
- Link to documentation pull request:

This is my first time working in HA Frontend, and also my first time working with Polymer/Lit. Please let me know of any best practices I may have missed and I'll be happy to amend the PR.

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @mattmattmatt, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;frontend#7223).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: src/dialogs/notifications/notification-drawer.js [hunk 1]
```diff
@@ -69,6 +75,13 @@ export class HuiNotificationDrawer extends EventsMixin(
               </div>
             </template>
           </dom-repeat>
+            <template is="dom-if" if="[[_moreThanOne(notifications)]]">
+              <div class="notification-actions">
+                <ha-progress-button raised="" on-click="_dismissAll">
```
Comment (zsarnett) [inline src/dialogs/notifications/notification-drawer.js:80]: does this need to be a progress button? The action is immediate not async

Comment (mattmattmatt) [reply src/dialogs/notifications/notification-drawer.js:80]: It might not need to. I'll see if I can find a more appropriate/simpler button.

Comment (mattmattmatt) [reply src/dialogs/notifications/notification-drawer.js]: Replaced it with the standard `mwc-button` (and fixed the indentation).

File: src/dialogs/notifications/notification-drawer.js [hunk 2]
```diff
@@ -111,10 +123,24 @@ export class HuiNotificationDrawer extends EventsMixin(
     this.open = false;
   }
 
+  _dismissAll(ev) {
+    ev.stopPropagation();
```
Comment (bramkragten) [inline src/dialogs/notifications/notification-drawer.js:127]: Is this needed?

**Review by bramkragten — APPROVED** [review]

File: src/dialogs/notifications/notification-drawer.js (hunk 2 above)
Comment (mattmattmatt) [reply src/dialogs/notifications/notification-drawer.js:127]: I followed the practice from somewhere else in the file. I don't think it's needed. Happy to remove it.

Comment (zsarnett) [reply src/dialogs/notifications/notification-drawer.js:127]:
```suggestion
```

**Pull request merged**